xformers/components/attention/nystrom.py

```diff
@@ -161,34 +166,41 @@ def forward(
         r"""
         key_padding_mask    Only a key padding mask is accepted here. The size must be (batch size, sequence length) or
                             (batch size * num_heads, 1, sequence length). If dimensions are not correct, the mask will
-                            be ignored.
+                            be ignored. An additive mask is expected, meaning float values using "-inf" to mask values
         """
+
         batched_dim = k.size(0)
         seq_len = k.size(-2)
+        tt = {"dtype": q.dtype, "device": q.device}
 
         if key_padding_mask is not None:
-            assert key_padding_mask.dtype == torch.bool
+            if key_padding_mask.dtype == torch.bool:
+                logging.warning(
+                    "Bool mask found, but an additive mask is expected. Converting but this is slow"
+                )
+                key_padding_mask = bool_mask_to_additive(key_padding_mask)
+
+            assert key_padding_mask is not None  # mypy is drunk
```

Comment:
I think if a return type is added to `bool_mask_to_additive` it may fix the mypy error

Reply:
ahh good point, I'll fix that, thank you !